spring-ai-core/src/main/java/org/springframework/ai/chat/client/observation/ChatClientInputContentObservationFilter.java

```diff
@@ -32,72 +31,54 @@
  */
 public class ChatClientInputContentObservationFilter implements ObservationFilter {
 
-	private static final KeyValue CHAT_CLIENT_SYSTEM_TEXT_NONE = KeyValue
-		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_SYSTEM_TEXT, KeyValue.NONE_VALUE);
-
-	private static final KeyValue CHAT_CLIENT_SYSTEM_PARAM_NONE = KeyValue
-		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_SYSTEM_PARAM, KeyValue.NONE_VALUE);
-
-	private static final KeyValue CHAT_CLIENT_USER_TEXT_NONE = KeyValue
-		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_USER_TEXT, KeyValue.NONE_VALUE);
-
-	private static final KeyValue CHAT_CLIENT_USER_PARAM_NONE = KeyValue
-		.of(ChatClientObservationDocumentation.HighCardinalityKeyNames.CHAT_CLIENT_USER_PARAMS, KeyValue.NONE_VALUE);
-
 	@Override
 	public Observation.Context map(Observation.Context context) {
 		if (!(context instanceof ChatClientObservationContext chatClientObservationContext)) {
 			return context;
 		}
 
-		chatClientObservationContext.addHighCardinalityKeyValue(chatClientSystemText(chatClientObservationContext))
-			.addHighCardinalityKeyValue(chatClientSystemParam(chatClientObservationContext))
-			.addHighCardinalityKeyValue(chatClientUserText(chatClientObservationContext))
-			.addHighCardinalityKeyValue(chatClientUserParam(chatClientObservationContext));
+		chatClientSystemText(chatClientObservationContext);
+		chatClientSystemParams(chatClientObservationContext);
+		chatClientUserText(chatClientObservationContext);
+		chatClientUserParams(chatClientObservationContext);
 
 		return chatClientObservationContext;
 	}
 
-	protected KeyValue chatClientSystemText(ChatClientObservationContext context) {
-		if (!StringUtils.hasText(context.getRequest().getUserText())) {
-			return CHAT_CLIENT_SYSTEM_TEXT_NONE;
+	protected void chatClientSystemText(ChatClientObservationContext context) {
+		if (!StringUtils.hasText(context.getRequest().getSystemText())) {
```

Comment:
Here the condition was checking the user text instead of the system text, I fixed it now

Reply:
good catch